Lib/sre_compile.py

```diff
@@ -249,17 +252,26 @@ def _compile_charset(charset, flags, code, fixup=None, fixes=None):
         else:
             raise error("internal: unsupported set operator %r" % (op,))
     emit(FAILURE)
+    return opcs
 
-def _optimize_charset(charset, fixup, fixes):
+def _optimize_charset(charset, flags, fixup, fixes):
     # internal: optimize character set
     out = []
     tail = []
     charmap = bytearray(256)
+    hascased = False
+    if fixup:
+        if flags & SRE_FLAG_UNICODE and not flags & SRE_FLAG_ASCII:
```

Comment:
Can the if/else statement from 264-267 be replaced with:

`iscased = _get_iscased(flags)`

?